lib/shared/files.js

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2018 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const _ = require('lodash')
+const path = require('path')
+const Bluebird = require('bluebird')
+const fs = Bluebird.promisifyAll(require('fs'))
+
+/* eslint-disable lodash/prefer-lodash-method */
+
+/**
+ * @summary Check if a file exists
+ * @function
+ * @public
+ *
+ * @param {String} fullpath - full path
+ * @returns {Boolean}
+ *
+ * @example
+ * console.log('I have pictures: ' + files.exists('~/Pictures'))
+ */
+exports.exists = (fullpath) => {
+  try {
+    return fs.existsSync(fullpath)
+  } catch (err) {
+    return err.code !== 'ENOENT'
+  }
+}
+
+/**
+ * @summary Get file metadata
+ * @function
+ * @private
+ *
+ * @param {String} dirname - directory name
+ * @param {String} [basename] - custom basename to append
+ * @returns {Object} file metadata
+ *
+ * @example
+ * const file = getFileMetadata('/home/user')
```

Comment:
What's the error handling case here? try / catch? Should that be illustrated in the example?